sky/backends/backend_utils.py

```diff
@@ -1822,26 +1837,26 @@ def run_ray_status_to_check_ray_cluster_healthy() -> bool:
                 raise exceptions.FetchClusterInfoError(
                     reason=exceptions.FetchClusterInfoError.Reason.HEAD)
             head_runner = runners[0]
-            rc, output, stderr = head_runner.run(
-                instance_setup.RAY_STATUS_WITH_SKY_RAY_PORT_COMMAND,
-                stream_logs=False,
-                require_outputs=True,
-                separate_stderr=True)
-            if rc:
-                raise RuntimeError(
-                    f'Refreshing status ({cluster_name!r}): Failed to check '
-                    f'ray cluster\'s healthiness with '
-                    f'{instance_setup.RAY_STATUS_WITH_SKY_RAY_PORT_COMMAND}.\n'
-                    f'-- stdout --\n{output}\n-- stderr --\n{stderr}')
 
-            ready_head, ready_workers = _count_healthy_nodes_from_ray(output)
+            ready_head, ready_workers, output, stderr = (
+                get_node_counts_from_ray_status(head_runner))
             total_nodes = handle.launched_nodes * handle.num_ips_per_node
-            if ready_head + ready_workers == total_nodes:
-                return True
-            raise RuntimeError(
-                f'Refreshing status ({cluster_name!r}): ray status not showing '
-                f'all nodes ({ready_head + ready_workers}/'
-                f'{total_nodes}); output: {output}; stderr: {stderr}')
+            if ready_head + ready_workers != total_nodes:
+                # if cluster JUST started, maybe not all the nodes have shown up.
+                # Wait for a few seconds and try again
+                logger.debug(f'Refreshing status ({cluster_name!r}): first '
+                             'ray status not showing all nodes '
+                             f'({ready_head + ready_workers}/{total_nodes}); '
+                             f'output:\n{output}\nstderr:\n{stderr}')
+                time.sleep(5)
+                ready_head, ready_workers, output, stderr = (
+                    get_node_counts_from_ray_status(head_runner))
+                if ready_head + ready_workers != total_nodes:
+                    raise RuntimeError(
+                        f'Refreshing status ({cluster_name!r}): ray status not '
+                        f'showing all nodes ({ready_head + ready_workers}/'
+                        f'{total_nodes}); output:\n{output}\nstderr:\n{stderr}')
```

Comment:
Can we only do such retry for clusters that are just launched? This is to make sure our refresh is responsive for those cluster that are actually preempted.

Reply:
For clusters that are actually preempted, we probably won't get to this point because the instances won't be running. I wanted to avoid the complexity since the case that this handles will rarely fail even for autostop.
That said it's pretty easy to add the check, so I will.

Reply:
offline discussion: just add a comment stating that it's rare to hit this